pkg/state/state_skel.go

```diff
@@ -91,32 +91,104 @@ func (s *stateSkel) updateObj(obj *unstructured.Unstructured) error {
 	return nil
 }
 
+// When updating, not all resources need an updated ResourceVersion
+func needToUpdateResourceVersion(kind string) bool {
+	if kind == "SecurityContextConstraints" ||
+		kind == "Service" ||
+		kind == "ServiceMonitor" ||
+		kind == "Route" ||
+		kind == "BuildConfig" ||
+		kind == "ImageStream" ||
+		kind == "PrometheusRule" {
+		return true
+	}
+	return false
+}
+
+func checkNestedFields(found bool, err error) {
+	if !found || err != nil {
+		return errors.Wrap(err, "nested field error")
+	}
+}
+
+func updateResourceVersion(req, found *unstructured.Unstructured) error {
+	kind := found.GetKind()
+
+	if needToUpdateResourceVersion(kind) {
+		version, fnd, err := unstructured.NestedString(found.Object, "metadata", "resourceVersion")
+		checkNestedFields(fnd, err)
```

Comment:
`checkNestedFields` will now silently fail, i don't think we want that right ?

As we (now) don't call `os.exit` i think an if statement is needed to check for errors.
either remove `checkNestedFields` and implement the logic directly or have it return an error and check it here.